JS/Lesson34/gulcin/recipe-app/src/modules/recipes/RecipeDetails/index.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react"
+import { Ingredients } from "./ingredients";
+import './styles.css';
+
+export const RecipeDetails = ({ id }) => {
+    const [recipe, setRecipe] = useState([]);
+
+    useEffect(() => {
+        fetch(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${id}`)
+            .then(res => res.json())
+            .then(data => {
+                if (data && data.meals) {
+                    setRecipe(data.meals);
+                }
+            }
+            ).catch(err => console.log(err));
+    }, [id])
+
+    return (
+        <section className="recipe-details">
```

Comment:
Since there may be no recipe information, it's good to have recipe details rendered conditionally:
```
{recipe ? (
<>
                        <h2>{recipe.strMeal.toUpperCase()}</h2>
                        <section className="img-ingredients">
                            <div className="recipe-ingredients">
                               <h3>Ingredients</h3>
                               <Ingredients recipe={recipe} />
                            </div>
                           <div className="recipe-img"><img alt={recipe.strMeal} src={recipe.strMealThumb} /></div>
                        </section>
                        <p className="recipe-text">{recipe.strInstructions}</p>
                    </>
):
<p>Loading...<p>
}
```

Instead of loading there could be an error component too, for example when we fail to find / fetch details. Feel free to add it if you like.

Reply:
Done.